cmd/serve.go

```diff
@@ -69,18 +79,42 @@ func cmdServe(cmd *cobra.Command, args []string) {
 		case <-ctx.Done():
 			return
 		case e := <-events:
-			switch e {
+			switch e.Type {
 			case kube.NewPodEvent:
-				metrics.GlobalPodMetrics.TotalNumberPods.Add(1)
+				podlog := log.New("pod", e.PodSelfLink)
+
+				cgPath := buildCgroupPath(cgroupRoot, e.PodUID, e.PodQOSClass)
+				podlog.Debug("new pod", "cgroup", cgPath)
+				metrics.TotalNum().Add(1)
+
+				bpfController, err := bpf.Attach(cgPath)
+				if err != nil {
+					podlog.Error("unable to install bpf program", "err", err)
+				}
+
+				podlog.Debug("set packets handler")
+				bpfController.SetPacketsHandler(func(v uint64) error {
+					metrics.SetOutgoingPackets(path.Base(e.PodSelfLink), float64(v))
+					return nil
+				})
+				podlog.Debug("starting bpf program")
+				go bpfController.Run(ctx)
+
 			case kube.DeletePodEvent:
-				metrics.GlobalPodMetrics.TotalNumberPods.Sub(1)
+				metrics.TotalNum().Sub(1)
+				log.Debug("pod gone", "pod", e.PodSelfLink, "cgroup", buildCgroupPath(cgroupRoot, e.PodUID, e.PodQOSClass))
```

Comment:
That was actually why I filed #7 because I didn't know if you had to do any cleanup work.

Will fix this now...